aten/src/ATen/native/Convolution.cpp

```diff
@@ -421,7 +421,7 @@ at::Tensor _convolution_nogroup(
           stride, padding, output_padding, dilation);
     } else if (dim == 5) {
       return at::thnn_conv_transpose3d(
-        input, weight, bias,
+        input, weight, kernel_size, bias,
```

Comment:
Kernel size can be inferred from the weghts right?

Reply:
That is true and is how it is done before. However, since this PR allows `gradWeight=NULL` the underlying THNN class needs a `kernel_size` parameter. Having a `kernel_size` is also consistent with all other THNN convolution functions.

Reply:
sorry but what does `gradWeight=NULL` has to do with this binding, which should call `updateOutput`, and should require weight to be defined?

Reply:
`kernel_size` needs to be added for `accGradParameters` to know weight sizes in case `gradWeight` is `NULL`. The way THNN bindings work is that it assumes `updateOutput`, `updateGradInput` and `accGradParameters` use a similar argument list. So I need to add `kernel_size` at all three places.

Reply:
I recall that @colesbury has been strongly advocating for excluding all those params from the signature. I'll leave it up to him to decide what to do in this case.

Reply:
@SsnL I don't think we need to store `kernel_size` there. In `accGradParameters`, we only need them to compute `outputSize`, but that can be obtained from `gradOutput`, right?

Reply:
@fmassa you are right. These values can be computed from output size. However, the original code uses kernel sizes to check the shape of `gradOutput`, and I don't want to relax the check. Furthermore, I don't understand why this is going to a big issue. Every other 3D THNN function I've seen that uses a kernel has these as input arguments.

Reply:
Discussed with @colesbury offline about this. This is fine because weight can be NULL. But in general, we should definitely avoid parameters that can already be inferred from other ones.